### PR TITLE
feat: remove window fetch mock and use MSW

### PR DESCRIPTION
**Pull request by marcosvega91**
**What**:

I Have removed the mock of `window.fetch` and I have used for react-fundamentals example `MSW`

**Why**:

Because we don't want to mock fetch anymore

**How**:

1) Update MSW to last release
2) Remove `window.fetch` mock from `setup-tests.js`
3) Use MSW in test and app

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [X] Ready to be merged

### CODE REVIEW

File: src/__tests__/__snapshots__/codegen.js.snap [hunk 1]
```diff
@@ -609,9 +642,12 @@ makeWorkshopApp({
     "src/final/09.js": () => import("./final/09.js"),
     "src/index.js": () => import("./index.js"),
     "src/react-app-env.d.ts": () => import("./react-app-env.d.ts"),
-    "src/server.js": () => import("./server.js"),
     "src/setupTests.js": () => import("./setupTests.js"),
-    "src/styles.css": () => import("./styles.css")
+    "src/styles.css": () => import("./styles.css"),
+    "src/test/server/dev-server.js": () => import("./test/server/dev-server.js"),
+    "src/test/server/handlers.js": () => import("./test/server/handlers.js"),
```
Comment (kentcdodds) [inline src/__tests__/__snapshots__/codegen.js.snap:648]: Let's update the codegen to ignore everything in the test directory

File: example/react-fundamentals/src/setupTests.js [hunk 2]
```diff
@@ -1 +1,6 @@
 import '@kentcdodds/react-workshop-app/setup-tests'
+import {server} from './test/server'
+
+beforeAll(() => server.listen())
```
Comment (kentcdodds) [inline example/react-fundamentals/src/setupTests.js:4]: Let's set onUnhandledRequest as "error" here.

Comment (marcosvega91) [reply example/react-fundamentals/src/setupTests.js]: I have added it, on both node and browser.

**Comment by kentcdodds** [comment]
Whoops! This probably should have been a breaking change. I'm not going to do anything about it though because the epic react workshops are the only users that I'm actively supporting 😅

**Comment by kentcdodds** [comment]
@all-contributors please add @marcosvega91 for code and tests

**Comment by marcosvega91** [comment]
Yes it can be managed 🚀🚀

**Pull request merged**